### DISCUSSION TITLE
let us create repos from the mobile app

### DISCUSSION THREAD

**Discussion by oa01** [community/community#5098] [Mobile]
And please more options for repo management in the mobile app

**Comment by HenSquared** [comment]
You can now create repositories on iOS and Android :) 
https://github.blog/changelog/2026-05-11-create-repositories-on-the-go-with-github-mobile/

**Comment by MahdiJDS** [comment]
Beyond just creating new repositories, extending the mobile app’s repository management capabilities would significantly enhance its utility for developers on the go.

Current State: While the app allows viewing and basic interaction, many essential management tasks still require a desktop or web interface.

Proposed Enhancements:

Create New Repository: (As previously requested) Ability to create a new repository with essential options like name, description, visibility (public/private), initialize with README, .gitignore, and license.

Edit Repository Settings:

Update repository name and description.
Toggle visibility (public/private).
Manage topics.
Edit the default branch name.
Enable/disable features like Wiki, Issues, Projects.
Branch Management:
View all branches.
Create new branches.
Set a default branch.
(Potentially) Delete branches (with clear warnings).
File/Content Management:
View file contents.
Edit text files directly within the app (similar to Gists).
Upload files to a repository.
Create new files and directories.
Delete files and directories (with confirmation).
Commit History & Details:
View commit history with author, date, and message.
View detailed changes for each commit.
Code Snippets/Gists Integration:
A seamless way to create or manage GitHub Gists from the mobile app, which can then be linked or embedded into repositories.
Benefits:

Full Mobile Workflow: Enables developers to manage their projects from anywhere, at any time.
Time Efficiency: Reduces the need to switch to a desktop for common administrative tasks.
Improved Responsiveness: Allows for quicker responses to issues, updates, or minor code changes.
Increased Engagement: Makes the mobile app a more central tool for developers, not just a viewer.
Implementing these features would transform the GitHub Android app into a robust mobile development companion.
